skaha/src/main/java/org/opencadc/skaha/SkahaAction.java

```diff
@@ -180,6 +192,29 @@ protected void initRequest() throws Exception {
         LocalAuthority localAuthority = new LocalAuthority();
         URI gmsSearchURI = localAuthority.getServiceURI("ivo://ivoa.net/std/GMS#search-0.1");
 
+        log.debug("using iam to fetch group details");
+        cacheGroupsWithIAM(subject, gmsSearchURI);
```

Comment:
You can delete the `cacheGroupsWithIAM` method and just put the code right in here.

The cacheGroupsWithForLDAP method can be deleted.

Reply:
I have deleted this cacheGroupsWithForLDAP method and also move the contents of cacheGroupsWithIAM function in the caller function